Rules/UseCompatibleTypes.cs

```diff
@@ -0,0 +1,854 @@
+            if (!commandAsts.Count().Equals(0))
+            {
+                List<dynamic> newObjectCommands = new List<dynamic>();
+                
+                foreach (CommandAst command in commandAsts)
+                {
+                    if (String.Equals(command.GetCommandName(), "New-Object", StringComparison.OrdinalIgnoreCase))
+                    {
+                        newObjectCommands.Add(command);
+                    }
+                }
+
+                addAstElementToList(newObjectCommands);
+            }
+            
+            // Types declared by the user (defined within a user-created class).
+            IEnumerable<Ast> definitionAsts = ast.FindAll(testAst => testAst is TypeDefinitionAst, true);
+            foreach (Ast definition in definitionAsts)
+            {
+                TypeDefinitionAst customType = (TypeDefinitionAst)definition;
+                customTypes.Add(customType.Name);
+            }
+
+            // These are namespaces used in the script.  This will help getting full type names for types
+            // from ast objects that only give us the type name as a string.  We will add these to the 
+            // beginning of our known namespaces list so we check those first.
+            IEnumerable<Ast> useStatementAsts = ast.FindAll(testAst => testAst is UsingStatementAst, true);
+            foreach (Ast use in useStatementAsts)
+            {
+                UsingStatementAst nspace = (UsingStatementAst)use;
+                knownNamespaces.Insert(0, nspace.Name.Value + ".");
+            }
+
+            // If we have no types to check, we can exit from this rule.
+            if (allTypesFromAst.Count.Equals(0))
+            {
+                return new DiagnosticRecord[0];
+            }
+
+            CheckCompatibility();
+
+            ProcessErrors();
+
+            return diagnosticRecords;
+        }
+
+        /// <summary>
+        /// Check if type is present in the target platform type list.
+        /// If not, create a Diagnostic Record for that type.
+        /// </summary>
+        private void CheckCompatibility()
+        {
+            foreach (dynamic typeObject in allTypesFromAst)
+            {
+                List<fullTypeNameObject> fullTypeNameObjectList = RetrieveFullTypeName(typeObject);
+
+                foreach (fullTypeNameObject nameObject in fullTypeNameObjectList)
+                {
+                    int couldNotResolveCount = 0;
+
+                    foreach (dynamic platform in psTypeMap)
+                    {
+                        if (nameObject.isCustomType)
+                        {
+                            // If this is a custom type, try and find it in our definition ast list.  If it's there
+                            // we can ignore it.
+                            if (customTypes.Contains(nameObject.fullName, StringComparer.OrdinalIgnoreCase))
```

Comment:
The `continue` here looks like it could be a level above since it does not depend on the innermost loop variable